python_modules/dagster/dagster/_core/decorator_utils.py

```diff
@@ -36,6 +51,11 @@ def get_function_params(fn: Callable[..., Any]) -> Sequence[Parameter]:
     return list(signature(fn).parameters.values())
 
 
+def get_type_hints(fn: Callable) -> Mapping[str, Any]:
+    target = fn.func if isinstance(fn, functools.partial) else fn
```

Comment:
is this just to deal with a difference in behavior between `typing_get_type_hints` and  `inspect`?

Reply:
yes, `get_type_hints` chokes on `partial`, `inspect.signature` doesn't